apps/o2-blocks/src/callout/editor.js

```diff
@@ -0,0 +1,349 @@
+/**
+ * Callout Block - Simplified Single Block Implementation
+ *
+ * A styled callout box with integrated icon, header, and content sections.
+ */
+
+import {
+	RichText,
+	InnerBlocks,
+	InspectorControls,
+	ColorPalette,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { registerBlockType } from '@wordpress/blocks';
+import { CustomSelectControl, PanelBody, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import metadata from './block.json';
+import {
+	CALLOUT_TEMPLATE_DEFAULT_COLOR,
+	CALLOUT_TEMPLATE_DEFAULT_BACKGROUND_COLOR,
+	CALLOUT_TEMPLATES,
+	CALLOUT_PANEL_COLORS,
+	CALLOUT_PANEL_BACKGROUND_COLORS,
+	CALLOUT_PANEL_DASHICONS,
+} from './constants';
+import { transforms } from './transform';
+
+import './editor.scss';
+import './view.scss';
+
+// === UTILITY FUNCTIONS ===
+
+const getIcon = ( templateName, customIcon = '' ) => {
+	if ( customIcon ) {
+		return customIcon.startsWith( 'dashicons-' ) ? customIcon : `dashicons-${ customIcon }`;
```

Comment:
what role do dashicons play in this? it seems strange to me that we would be passing around custom icons that may or may not have a `dashicons-` prefix.